mlflow/projects/backend/loader.py

```diff
@@ -1,13 +1,24 @@
 import entrypoints
 import logging
 
+from mlflow.projects.backend.local import LocalBackend
+
+
 ENTRYPOINT_GROUP_NAME = "mlflow.project_backend"
 
 __logger__ = logging.getLogger(__name__)
 
 
 # Statically register backend defined in mlflow
-MLFLOW_BACKENDS = dict()
+MLFLOW_BACKENDS = {
+    'local': LocalBackend,
+}
+
+try:
```

Comment:
I'm guessing we perform this check to avoid `ModuleNotFoundErrors` when e.g. the `kubernetes` Python library isn't available. It makes sense to avoid such errors, but could we instead try to `import kubernetes` within the actual `run` method of the `KubernetesBackend` (so that we only throw a `ModuleNotFoundError` if you try to actually run a k8s project)? It seems a bit odd to not register the kubernetes backend at all if the `kubernetes` python library isn't present, as this will cause the backend name to not appear in help documentation / the list of supported backends generated [here](https://github.com/mlflow/mlflow/pull/2934/files#diff-16104674981d54b31e320402a143eb19R101)